### PR TITLE
Add Dockerfiles for ARM

### PR DESCRIPTION
**Pull request by elezar**
This PR adds docker files for ARM architectures like the raspberry pi.

I have added Alpine and Debian-based images for `v0.14` and `v1.2`.

*Notes*:
* The cross-platform build (e.g. building locally on Docker for Mac) doesn't currently work for the Alpine images
* I switched to apt packages for `gosu` and `dumb-init` for 
* I have tested the cross-platform build of the v1.2 Debian image
* I have tested the native ARM build of the v1.2 Alpine onbuild image
* I use the v0.14 ARM Alpine image (built on a Raspberry Pi) in a Raspberry PI speedtest project:
    ```bash
    docker build --build-arg CROSS_BUILD_START=":" --build-arg CROSS_BUILD_END=":" -t elezar/fluentd:v0.14.14-armhf-onbuild .
    ```
    Note this sets the cross-build start and end commands to no-ops.

### CODE REVIEW

**Comment by repeatedly** [comment]
v0.14 images are not needed because v0.14 is older version of v1.x.

**Comment by repeatedly** [comment]
And please add your sign with `-s` to commits for DCO.

**Comment by elezar** [comment]
@repeatedly thanks for the feedback. I have amended the commits and also removed the v0.14 ARM images.

File: v0.12/alpine-onbuild/Dockerfile [hunk 1]
```diff
@@ -9,8 +9,6 @@ ENV DUMB_INIT_VERSION=1.2.0
 
 ENV SU_EXEC_VERSION=0.2
 
-ARG DEBIAN_FRONTEND=noninteractive
```
Comment (repeatedly) [inline v0.12/alpine-onbuild/Dockerfile]: Is this deletion intented?

**Comment by repeatedly** [comment]
Thanks. Just merged!

**Comment by repeatedly** [comment]
Building ARM image on Docker hub failed by `exec format error`.
Do you have an idea for resolving this problem? > https://hub.docker.com/r/fluent/fluentd/builds/bxpxw6mvvwn4ymsdr3atugb/

**Comment by elezar** [comment]
Hmmm. Does docker provide native ARM support in some way? I tested on Docker for Mac, but will do some checking to see what could be the cause and open a new PR if I find anything.

**Comment by elezar** [comment]
I have found docker/hub-feedback#1261which may provide what we need.

**Pull request merged**